tests/be-golang-plain/jenkinsfile_test.go

```diff
@@ -0,0 +1,129 @@
+
+	// cleanup and create bb resources for this test
+	utils.CleanupAndCreateBitbucketProjectAndRepo(
+		quickstarterName, componentId)
+
+	// run provision job for quickstarter in project's cd jenkins
+	stages, err := utils.RunJenkinsFile(
+		"ods-quickstarters",
+		values["ODS_BITBUCKET_PROJECT"],
+		values["ODS_GIT_REF"],
+		coreUtils.PROJECT_NAME,
+		fmt.Sprintf("%s/Jenkinsfile", quickstarterName),
+		coreUtils.PROJECT_NAME_CD,
+		coreUtils.EnvPair{
+			Name:  "COMPONENT_ID",
+			Value: componentId,
+		},
+		coreUtils.EnvPair{
+			Name:  "GIT_URL_HTTP",
+			Value: fmt.Sprintf("%s/%s/%s.git", values["REPO_BASE"], coreUtils.PROJECT_NAME, componentId),
+		},
+	)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	fmt.Printf("Provision Build for %s returned:\n%s", componentId, stages)
+
+	// verify provision jenkins stages - against golden record
+	expected, err := ioutil.ReadFile("golden/jenkins-provision-stages.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expectedAsString := string(expected)
+	if stages != expectedAsString {
+		t.Fatalf("Actual jenkins stages from prov run: %s don't match -golden:\n'%s'\n-jenkins response:\n'%s'",
+			componentId, expectedAsString, stages)
+	}
+
+	// run master build of provisioned quickstarter in project's cd jenkins
+	stages, err = utils.RunJenkinsFile(
+		componentId,
+		coreUtils.PROJECT_NAME,
+		"master",
+		coreUtils.PROJECT_NAME,
+		"Jenkinsfile",
+		coreUtils.PROJECT_NAME_CD,
+		coreUtils.EnvPair{
+			Name:  "COMPONENT_ID",
+			Value: componentId,
+		},
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	fmt.Printf("Master (code) build for %s returned:\n%s", componentId, stages)
+
+	// verify run and build jenkins stages - against golden record
+	expected, err = ioutil.ReadFile("golden/jenkins-build-stages.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expectedAsString = string(expected)
+	if stages != expectedAsString {
+		t.Fatalf("Actual jenkins stages from build run: %s don't match -golden:\n'%s'\n-jenkins response:\n'%s'",
+			componentId, expectedAsString, stages)
+	}
+
+	// sonar scan check
+	sonarscan, err := utils.RetrieveSonarScan(
+		fmt.Sprintf("%s-%s", coreUtils.PROJECT_NAME, componentId))
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// verify sonar scan - against golden record
+	expected, err = ioutil.ReadFile("golden/sonar-scan.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expectedAsString = string(expected)
+	if sonarscan != expectedAsString {
+		t.Fatalf("Actual sonar scan for run: %s doesn't match -golden:\n'%s'\n-sonar response:\n'%s'",
+			componentId, expectedAsString, sonarscan)
+	}
+
+	resourcesInTest := coreUtils.Resources{
+		Namespace:         coreUtils.PROJECT_NAME_DEV,
+		ImageTags:         []coreUtils.ImageTag{{Name: componentId, Tag: "latest"}},
+		BuildConfigs:      []string{componentId},
+		DeploymentConfigs: []string{componentId},
+		Services:          []string{componentId},
+		ImageStreams:      []string{componentId},
+	}
+
+	coreUtils.CheckResources(resourcesInTest, t)
+
+}
```

Comment:
@clemensutschig Wondering - did you decide on purpose against a table test that just runs the same code against different directories / golden files? The source code here looks largely identical to the other quickstarters ...

Reply:
@michaelsauter i thought about this as well.. but as quickstarters grow and Change i fehlt the couple lines of code dont matter..

Reply:
I see. However, right now I fail to see how we actually would need different code:

* Provisioning is a standardised process - calling a Jenkins Pipeline with some params. Right now the params are fixed, but even if they are more flexible, it would be easy to cover that in a table driven test
* Building a component is a standardised process - calling a Jenkins Pipeline with no params
* As output, there will always be stages as long as we use Jenkins, and SonarQube (if applicable to the QS) also returns the same thing for every QS (as every QS is equal to a SQ project)

The only thing were quickstarters could differ:
* Created resources
* Created files in the resulting repository

Those would be easy to cover in table driven tests. If not enough, we could even pass functions that do special things in the table driven tests to retain complete flexibility ...

I think we should think about it - as, at least right now, a "QS test" would only need to be:
* a name
* golden files

Would be easier to create / maintain?